apm-agent-core/src/test/java/co/elastic/apm/agent/testutils/assertions/metrics/MetricsetJson.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.testutils.assertions.metrics;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.Map;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class MetricsetJson {
```

Comment:
Good idea to use jackson object mapping to make tests easier, however I think that having more "explicit" mapping of the attributes is more robust to object attributes renaming (when it happens it's a pain to debug). Alternatively, can we make jackson throw an error when a field in JSON is not mapped to this object ?

Reply:
>Alternatively, can we make jackson throw an error when a field in JSON is not mapped to this object

That is actually the default behaviour of jackson. For this reason I added the `@JsonIgnoreProperties(ignoreUnknown = true)` here, but you are right, we should be more explicit: I will replace `ignoreUnknown=true` with `value` with an explicit list of allowed attribute names taken from the metricset spec.
This way, we will get an error if any unknown attribute is present.